### PR TITLE
mds: dump minimal cli output during recursive scrub trigger

### PR DESCRIPTION
**Pull request by vshankar**
Signed-off-by: Venky Shankar <vshankar@redhat.com>
Fixes: http://tracker.ceph.com/issues/23958

### CODE REVIEW

**Comment by vshankar** [comment]
At the time of recursive scrub trigger, looks like there isn't much information to dump other than just dumping a valid json.

**Comment by ukernel** [comment]
maybe add a message " queued"

**Comment by vshankar** [comment]
> maybe add a message " queued"

I had thought of adding `async_scheduled : true` in the json output, but couldn't find a pressing need to have that.

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -12309,6 +12309,13 @@ void MDCache::enqueue_scrub(
 
   mdr->internal_op_finish = cs;
   enqueue_scrub_work(mdr);
+
+  // since recursive scrub is asynchronous, dump minimal output
+  // to not upset cli tools.
+  if (recursive && tag.empty()) {
```
Comment (batrick) [inline src/mds/MDCache.cc]: What's the output when the `!tag.empty()`? Should they be similar?

Something to think about: assigning a UUID to the tag if it's undefined.

Comment (vshankar) [reply src/mds/MDCache.cc]: `tag path` admin socket command passes a non-empty tag -- we can avoid having that check here is that's what is needed.

Comment (vshankar) [reply src/mds/MDCache.cc]: > What's the output when the !tag.empty()? Should they be similar?

PR updated for dumping minimal output for `scrub_path` and `tag path` asok commands.

> Something to think about: assigning a UUID to the tag if it's undefined.

Yes, maybe. Right now the tags seem to be only used during `ceph-data-scan` tool.

**Comment by vshankar** [comment]
@batrick fixed and updated -- test cases pass when run with `vstart_runner`.

Note than I also added the checks in `qa/tasks/cephfs/test_scrub.py`.

File: qa/tasks/cephfs/test_forward_scrub.py [hunk 2]
```diff
@@ -232,7 +232,8 @@ def test_inotable_sync(self):
         self.mount_a.umount_wait()
 
         with self.assert_cluster_log("inode table repaired", invert_match=True):
-            self.fs.mds_asok(["scrub_path", "/", "repair", "recursive"])
+            out_json = self.fs.mds_asok(["scrub_path", "/", "repair", "recursive"])
+            self.assertEqual(out_json, {})
```
Comment (batrick) [inline qa/tasks/cephfs/test_forward_scrub.py]: When we eventually fill in this JSON object with useful output, the test will unnecessarily break. I think it's enough to confirm `out_json is not None`.

**Pull request merged**